dev_tools/nyaml2nxdl/README.md

```diff
@@ -44,15 +44,49 @@ Options:
 * Optionality: For all fields, groups and attributes in `application definitions` are `required` by default, except anything (`recommended` or `optional`) mentioned.
 
 **Special keywords**: Several keywords can be used as childs of groups, fields, and attributes to specify the members of these. Groups, fields and attributes are nodes of the XML tree.
-* **doc**: A human-readable description/docstring
+* **doc**: 
+   - A human-readable description/docstring
+   - Doc string may also come as a list of doc parts when user wants to add `reference` for a concept.
+      ```yaml
+         energy:  # field
+            doc:
+               - | 
+               Part1 of the entire doc.
+               part1 of the entire doc.
+               - |  # Reference for concept
+               "xref:   
+                  spec: <spec>
+                  term: <term>
+                  url: <url>"
+               - | 
+               Rest of the doc
+               rest of the doc           
```

Comment:
It should also explain how to write a simple docstring without xref.

Reply:
updated.